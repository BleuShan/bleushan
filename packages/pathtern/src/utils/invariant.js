/* eslint-disable fp/no-nil */
/* eslint-disable better/explicit-return */
/* eslint-disable fp/no-mutation */
/* eslint-disable better/no-ifs */
/* eslint-disable fp/no-throw */
export default function invariant(
  condition,
  message,
  name = 'Invariant Violation'
) {
  if (condition) {
    const error = Reflect.construct(Error, [message])
    error.name = name
    throw error
  }
}
