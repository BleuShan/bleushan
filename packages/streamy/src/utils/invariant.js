import {typeOf} from './typeOf.js'

export function invariant({condition, message, errorConstructor = Error, args = []} = {}) {
  if (typeOf(errorConstructor) !== 'function') {
    throw new TypeError(`errorContructor is not a function`)
  }

  const msgType = typeOf(message)
  if (msgType !== 'string') {
    throw new TypeError(`expected message to be a string, received: ${message}`)
  }

  const conditionResult = typeOf(condition) === 'function' ? condition() : condition

  const condType = typeOf(conditionResult)
  if (condType !== 'boolean') {
    throw new TypeError(
      `expected condition be able to be evaluated to a boolean, received: ${conditionResult}`
    )
  }

  if (!conditionResult) {
    throw Reflect.construct(errorConstructor, [message, ...args])
  }
}
