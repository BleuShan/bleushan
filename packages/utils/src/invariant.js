import typeOf from './typeOf.js'
import isFunction from './isFunction.js'
import isPlainObject from './isPlainObject.js'

export default function invariant(options) {
  if (!isPlainObject(options)) {
    throw new TypeError(`Expected options to be an object, received: ${options}`)
  }
  const {condition, message, errorType = Error, args = []} = options
  if (!isFunction(errorType)) {
    throw new TypeError(`options.errorType is not a function`)
  }

  const msgType = typeOf(message)
  if (msgType !== 'string') {
    throw new TypeError(`Expected options.message to be a string, received: ${message}`)
  }

  const conditionResult = isFunction(condition) ? condition() : condition

  const condType = typeOf(conditionResult)
  if (condType !== 'boolean') {
    throw new TypeError(
      `Expected options.condition be able to be evaluated to a boolean, received: ${conditionResult}`
    )
  }

  if (!conditionResult) {
    throw Reflect.construct(errorType, [message, ...args])
  }
}
