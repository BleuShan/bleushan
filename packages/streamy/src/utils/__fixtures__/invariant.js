import {invariant} from '../invariant.js'

export function invariantWhenInvokedWith(configuration) {
  return function invoke() {
    invariant(configuration)
  }
}
