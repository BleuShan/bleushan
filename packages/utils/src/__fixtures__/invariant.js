import invariant from '../invariant.js'

export default function invariantWhenInvokedWith(configuration) {
  return function invoke() {
    invariant(configuration)
  }
}
