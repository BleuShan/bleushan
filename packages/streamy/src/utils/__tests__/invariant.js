import {invariant} from '../invariant.js'

describe('invariant', () => {
  it('should throw an when message is invalid', () => {
    expect(() => {
      invariant({message: 1})
    }).toThrowErrorMatchingSnapshot()
  })

  it('should throw an when conditon is a function that returns a string', () => {
    expect(() => {
      invariant({message: 'string', condition: () => 'ste'})
    }).toThrowErrorMatchingSnapshot()
  })

  it('should throw an error when conditon is a string', () => {
    expect(() => {
      invariant({message: 'string', condition: ''})
    }).toThrowErrorMatchingSnapshot()
  })

  it('should throw an error when the errorConstructor is a string', () => {
    expect(() => {
      invariant({message: 'string', condition: false, errorConstructor: 'stes'})
    }).toThrowErrorMatchingSnapshot()
  })
})
