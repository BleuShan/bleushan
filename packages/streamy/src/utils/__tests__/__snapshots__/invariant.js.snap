// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invariant should throw an error when conditon is a string 1`] = `"expected condition be able to be evaluated to a boolean, received: "`;

exports[`invariant should throw an error when the errorConstructor is a string 1`] = `"errorContructor is not a function"`;

exports[`invariant should throw an when conditon is a function that returns a string 1`] = `"expected condition be able to be evaluated to a boolean, received: ste"`;

exports[`invariant should throw an when message is invalid 1`] = `"expected message to be a string, received: 1"`;
